test/unit/test_grouped_loader.py

```diff
@@ -0,0 +1,238 @@
+
+FileLoaderDict = dict(nifti=NiftiFileLoader, h5=H5FileLoader)
+nifti_path = join("data", "test", "nifti", "grouped")
+h5_path = join("data", "test", "h5", "grouped")
+DataPaths = dict(nifti=nifti_path, h5=h5_path)
+
+
+def test_validate_data_files():
+    """
+    Test validate_data_files function looks for inconsistencies in the fixed/moving image and label lists.
+    If there is any issue it will raise an error, otherwise it returns None.
+    """
+    for key_file_loader, file_loader in FileLoaderDict.items():
+        for train_split in ["train", "test"]:
+            for labeled in ["True", "False", 1, 0]:
+                data_dir_path = join(DataPaths[key_file_loader], train_split)
+                image_shape = (64, 64, 60)
+                common_args = dict(
+                    file_loader=file_loader,
+                    labeled=labeled,
+                    sample_label="all",
+                    intra_group_prob=1,
+                    intra_group_option="forward",
+                    sample_image_in_group=False,
+                    seed=None if train_split == "train" else 0,
+                )
+
+                data_loader = GroupedDataLoader(
+                    data_dir_path=data_dir_path, image_shape=image_shape, **common_args
+                )
+
+                assert data_loader.validate_data_files() is None
+
+
+def test_num_intra_group_probs():
+    """
+    Test whether fewer than two groups throws an error for intra_group_prob < 1
+     - data/test/h5/grouped/test contains only a single group
+    """
+    with pytest.raises(Exception) as e_info:
+        data_dir_path = join(DataPaths["h5"], "test")
+        image_shape = (64, 64, 60)
+        common_args = dict(
+            file_loader=H5FileLoader,
+            labeled=True,
+            sample_label="all",
+            intra_group_prob=0.5,
+            intra_group_option="forward",
+            sample_image_in_group=False,
+            seed=0,
+        )
+        data_loader = GroupedDataLoader(
+            data_dir_path=data_dir_path, image_shape=image_shape, **common_args
+        )
+        data_loader.close()
+
+        assert "we need at least two groups" in str(e_info.value)
+
+
+def test_sample_indices_intra():
+    """
+    Test the number of samples and indices are correct for a single image sample
+    """
+    data_dir_path = join(DataPaths["h5"], "train")
+    image_shape = (64, 64, 60)
+    common_args = dict(
+        file_loader=FileLoaderDict["h5"],
+        labeled=True,
+        sample_label="all",
+        intra_group_prob=0.5,
+        intra_group_option="forward",
+        sample_image_in_group=True,
+        seed=None,
+    )
+    data_loader = GroupedDataLoader(
+        data_dir_path=data_dir_path, image_shape=image_shape, **common_args
+    )
+
+    assert data_loader.sample_indices is None
+    assert data_loader._num_samples == 2
+
+
+def test_group_indices_inter():
```

Comment:
you are testing `get_inter_sample_indices` here, maybe rename the test function name to `test_get_inter_sample_indices`?

In the following docstring, we are testing inter group sampling with all possibilities. "entire group is sampled" is not that accurate.

Reply:
Amended